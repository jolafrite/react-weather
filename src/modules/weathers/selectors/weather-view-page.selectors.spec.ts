import {
  getViewPageState,
  getViewPageLoaded,
  getViewPageLoading,
  getViewPageError,
  getViewPageWeather,
  getViewPageId,
  getViewPageForecast,
  getViewPageTodayHourlyForecast,
  getViewPageNextDaysHourlyForecast,
  getViewPageNextDaysForecast
} from "./weather-view-page.selectors";

describe("selectors: weather view page", () => {
  describe("getViewPageState", () => {
    test("should return the weather viewPage state", () => {
      const state = {
        weathers: {
          viewPage: {
            state: "weathers viewPage"
          }
        }
      };

      expect(getViewPageState(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageLoaded", () => {
    test("should return if the status is loaded", () => {
      const state = {
        weathers: {
          viewPage: {
            status: "loaded"
          }
        }
      };

      expect(getViewPageLoaded(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageLoading", () => {
    test("should return if the status is loading", () => {
      const state = {
        weathers: {
          viewPage: {
            status: "loading"
          }
        }
      };

      expect(getViewPageLoading(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageError", () => {
    test("should return if the status is error", () => {
      const state = {
        weathers: {
          viewPage: {
            status: "error"
          }
        }
      };

      expect(getViewPageError(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageId", () => {
    test("should return the id", () => {
      const state = {
        weathers: {
          viewPage: {
            id: 1234
          }
        }
      };

      expect(getViewPageId(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageWeather", () => {
    test("should return the matching weather", () => {
      const state = {
        weathers: {
          weathers: {
            entries: {
              1234: "weather from state"
            }
          },
          viewPage: {
            id: 1234
          }
        }
      };

      expect(getViewPageWeather(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageForecast", () => {
    test("should return the matching forecast", () => {
      const state = {
        weathers: {
          forecasts: {
            entries: {
              1234: "forecast from state"
            }
          },
          viewPage: {
            id: 1234
          }
        }
      };

      expect(getViewPageForecast(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageTodayHourlyForecast", () => {
    test("should return the weather today`s hourly forecast", () => {
      const state = {
        weathers: {
          weathers: {
            entries: {
              1234: { date: "2019-10-30" }
            }
          },
          forecasts: {
            entries: {
              1234: {
                "2019-10-30": "today forecast from state"
              }
            }
          },
          viewPage: {
            id: 1234
          }
        }
      };

      expect(getViewPageTodayHourlyForecast(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageNextDaysHourlyForecast", () => {
    test("should return the weather next days hourly forecast", () => {
      const state = {
        weathers: {
          weathers: {
            entries: {
              1234: { date: "2019-10-10" }
            }
          },
          forecasts: {
            entries: {
              1234: {
                "2019-10-09": "yesterday forecast from state",
                "2019-10-10": "today forecast from state",
                "2019-10-11": "tomorrow forecast from state",
                "2019-10-12": "after tomorrow forecast from state"
              }
            }
          },
          viewPage: {
            id: 1234
          }
        }
      };

      expect(getViewPageNextDaysHourlyForecast(state as any)).toMatchSnapshot();
    });
  });

  describe("getViewPageNextDaysForecast", () => {
    test("should return the predicted next days forecast", () => {
      const state = {
        weathers: {
          weathers: {
            entries: {
              1234: { date: "2019-10-10" }
            }
          },
          forecasts: {
            entries: {
              1234: {
                "2019-10-10": "today forecast from state",
                "2019-10-11": {
                  "09:00:00": {
                    main: { temp: 21, temp_max: 25, temp_min: 19 },
                    weather: {
                      id: 803,
                      main: "Clouds",
                      description: "broken clouds",
                      icon: "04n"
                    }
                  },
                  "10:00:00": {
                    main: { temp: 17, temp_max: 19, temp_min: 15 },
                    weather: {
                      id: 701,
                      main: "Mist",
                      description: "mist",
                      icon: "50d"
                    }
                  },
                  "11:00:00": {
                    main: { temp: 23, temp_max: 24, temp_min: 22 },
                    weather: {
                      id: 300,
                      main: "Drizzle",
                      description: "light intensity drizzle",
                      icon: "09d"
                    }
                  }
                },
                "2019-10-12": {
                  "09:00:00": {
                    main: { temp: -10, temp_max: -5, temp_min: -12 },
                    weather: {
                      id: 802,
                      main: "Clouds",
                      description: "scattered clouds",
                      icon: "03d"
                    }
                  },
                  "10:00:00": {
                    main: { temp: 5, temp_max: 7, temp_min: 2 },
                    weather: {
                      id: 800,
                      main: "Clear",
                      description: "clear sky",
                      icon: "01d"
                    }
                  },
                  "11:00:00": {
                    main: { temp: -3, temp_max: 0, temp_min: -5 },
                    weather: {
                      id: 804,
                      main: "Clouds",
                      description: "overcast clouds",
                      icon: "04d"
                    }
                  }
                }
              }
            }
          },
          viewPage: {
            id: 1234
          }
        }
      };

      expect(getViewPageNextDaysForecast(state as any)).toMatchSnapshot();
    });
  });
});
