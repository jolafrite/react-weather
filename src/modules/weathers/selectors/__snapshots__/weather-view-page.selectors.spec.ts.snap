// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors: weather view page getViewLast5DaysForecast should return the min_temp max_temp and the average temp of the day 1`] = `
Object {
  "2014-07-23": Object {
    "hourly": Object {
      "09:00:00": Object {
        "temp": 21,
        "temp_max": 25,
        "temp_min": 19,
      },
      "10:00:00": Object {
        "temp": 17,
        "temp_max": 19,
        "temp_min": 15,
      },
      "11:00:00": Object {
        "temp": 23,
        "temp_max": 24,
        "temp_min": 22,
      },
    },
    "temp": 20.33,
    "temp_max": 23,
    "temp_min": 17,
  },
  "2014-07-24": Object {
    "hourly": Object {
      "09:00:00": Object {
        "temp": -10,
        "temp_max": -5,
        "temp_min": -12,
      },
      "10:00:00": Object {
        "temp": 5,
        "temp_max": 7,
        "temp_min": 2,
      },
      "11:00:00": Object {
        "temp": -3,
        "temp_max": 0,
        "temp_min": -5,
      },
    },
    "temp": -2.67,
    "temp_max": 5,
    "temp_min": -10,
  },
}
`;

exports[`selectors: weather view page getViewPageError should return if the status is error 1`] = `true`;

exports[`selectors: weather view page getViewPageLoaded should return if the status is loaded 1`] = `true`;

exports[`selectors: weather view page getViewPageLoading should return if the status is loading 1`] = `true`;

exports[`selectors: weather view page getViewPageState should return the weather viewPage state 1`] = `
Object {
  "state": "weathers viewPage",
}
`;

exports[`selectors: weather view page getViewPageWeather should return the matching weather 1`] = `
Object {
  "id": 1234,
}
`;

exports[`selectors: weather view page getViewPageWeatherId should return the id 1`] = `1234`;
